### DISCUSSION TITLE
Welcome to the community, come here and say hi 👋  !

### DISCUSSION THREAD

**Discussion by christian-bromann** [webdriverio/webdriverio#6217] [Community]
Hey you 👋 !

I am excited that you have found the way deep into the internet to this thread. The WebdriverIO team wants to welcome you to the community! Here are some important links you might want to look into:

- You would like to contribute to the project? Check out some of our [`good first pick`](https://github.com/webdriverio/webdriverio/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22good+first+pick%22) issues and have a look into our [contribution guidelines](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md).
- You have a question on how to use WebdriverIO? Feel free to join our community chat [on Gitter](https://gitter.im/webdriverio/webdriverio) or post a Q&A question in the [discussions forum](https://github.com/webdriverio/webdriverio/discussions/new)
- You experience a bug and have a concrete reproducible example? Drop us [an issue](https://github.com/webdriverio/webdriverio/issues/new?template=--bug-report.md) and we take it from there 😉 

Aside from that we would like to get to know you and the people using WebdriverIO. Feel free to drop us a line how you use the project and what you enjoy about it 😊 

Let's start with me: Hi! I am Christian working in the Open Source Program Office at [Sauce Labs](https://github.com/saucelabs). I started using WebdriverIO a long time ago when it was called WebdriverJS. I used it to test a Backbone.js application (yes, that's a framework people used when there was no React). I really enjoyed using it and started contributing to the project. At some point the original creator of the project (@camme god bless you 🙏 ) made me an official contributor. From there I used the project to learn everything about Node.js and Open Source. Eventually it got me a job at Sauce Labs where I deepened my interest and knowledge on automated testing practices. These days I am fortunate to maintain the project as part of my job. It is very rewarding to see how many people are using it on a daily basis and how many peoples lives were influenced by it. There is a lot more to come and I am excited about this projects future 🚀

Enough about me, tell me about you!

**Comment by alisman** [comment]
Greeting!  We have been using webdriver.io at cBioPortal (c for Cancer) a visualization platform for cancer genomic data that is used by cancer researchers and medical professionals around the world.    We've been on v4 for three years.  Now trying to upgrade to v6.  Keep up the good work!

**Reply by alisman** [reply]
I am excited to try the react component selectors.   A couple ideas to throw out there from our experience:
1. Support for jQuery selectors.  Seems like it would be a cinch to just do the execution in browser and return ref to wdio scope
2. Support for disabling mouse.  We do this by injecting an transparent layer above the entire interface while we take a screenshot.  This prevents a lot of flakiness.

**Reply by christian-bromann** [reply]
Welcome @alisman 👋 

what is so flaky about making screenshots with mouse enabled?

**Reply by alisman** [reply]
we find that hover events often get triggered in a flaky way.  we have a lot of little UI buttons that only show up on hover, and they will be randonly showing or not showing from run to run.   we tried moving mouse off screen.  doesn't work.  but my latest attempt which basically prevents any hover interaction with underlying site seems to work.

**Comment by starbuck251** [comment]
Hi, I am Zoe, live/work in the UK ... after using Watir for a while I moved job and started as QA Manager at a place where we make software used by the NHS whereby we are replacing paper forms/notes etc with electronic documents at a growing company.  I had to look into Javascript automation frameworks and as we needed to be cross browser compatible (since the NHS love their IE11!) and found WDIO. 
I loved the simplicity of the commands and after building up a few trial tests using cucumber on a POC I now have 2 other testers (who had more of a Java background)on the team and we are smashing our way through a LOT of regression tests and all of us learning more each day about Javascript and WebdriverIO automation. I am quite a fan of it now and love building tests!

**Comment by amrsa1** [comment]
I would like to suggest to see some newsletter whenever there is a major WDIO release or important updates, it will be good refreshment for WDIO users to be following up with updates and a good way of promoting the framework, I have seen that so often in cypress and it's really useful and I think it would be amazing to have it as well.

**Reply by christian-bromann** [reply]
Thanks for the suggestion @Amrkamel1! It is definitely a great idea. Given that this project can't afford to pay a marketing team we are depending on contributions from people in the community (like you 😉). If you would like to help us on this effort let's chat!

Would you like to introduce yourself as well?

**Reply by amrsa1** [reply]
Ah sorry forgot to introduce myself,

Hi all, I'm Amr work as an automation engineer for an IT company beside several freelancing projects, and I have been trying almost all existing Automation fromwork for both web and mobile and I have experienced the pros and cons for each until I stumbled over WDIO framework :D , this framework really offers a lot of features and services that make handling the drivers way easier than some other frameworks, also support team is so responsive and helpful I'm glad to see the framework advance even further.

@christian-bromann  definitely we can chat about it, I will reach out.

**Comment by goveanick** [comment]
Hello! My name is Nick and I work at a company called [Software Advice](https://www.softwareadvice.com/). We're a company that helps small to medium sized businesses find the right software for their business. We've been using WDIO since version 4 if I'm not mistaken. I work on a team that consists of 4 QA engineers in total and we've all come to love WDIO. I personally really like all the resources that you guys have created from the docs, the gitter channel, and the big community that backs the project. 

A little bit about how we use WDIO, we have multiple repos consisting of different technologies where we're using WDIO. 

- We have some php repos where we've integrated WDIO inside of 
- We have some node js repos where we've also integrated WDIO
- We also have some react js repos where we use WDIO
- We also use browserstack with all of our WDIO integrations
- We also run all our tests across multiple browsers using WDIO - (Chrome, Safari, FF, Edge, IE)
- We also run some tests on mobile devices, although not nearly as many as we'd like. We hope to build this out further

One thing I'd like to see is more resources/examples on how WDIO can be used in AWS. Not really anything in particular, but I've seen a few articles here and there about running tests in lambdas and would love to see more of that kind of stuff.

**Comment by pmerwin** [comment]
Hey my dudes, my name is Phil and I am a SDET at cbtnuggets.com, and have been the automation lead for browser testing and a huge advocate for WDIO since vs 4.  WDIO offers some awesome features, and allows synchronous test execution that helps to simplify test writing and maintenance for us.  The wdio-intercept, mocking, and devtools services have allowed us to add some interesting and robust test logic to our end to end tests, and the devtools performance domain allows us to run Lighthouse tests on all of our routes every night (we run on chromium in AWS lambda for these tests with WDIO), which has proved invaluable and has allowed us to really improve and maintain our scores.  The wdio saucelabs service has allowed us to do multi browser tests and integrates seamlessly with Sauce Labs pretty much out of the box. WDIO is awesome and allows our engineers to use one editor all day long as we are a node shop.

**Comment by cg-tester** [comment]
@starbuck251 
I want to ask one question. 
should I remove code  services:['chromedriver'] in wdio.conf.js file ?

**Comment by MahadevMG** [comment]
Hi, I’m Mahadev from Bangalore, India. I work as a Senior QA Automation Engineer and have been extensively using WebDriverIO. Transitioning from Selenium and Java to WebDriverIO has been fantastic; I’m really impressed with its simplicity and ease of use. It has become my go-to tool for various test frameworks in our organization, and I’m excited to continue this journey with WebDriverIO. Thanks!

**Comment by alwinjoseph02** [comment]
Hi, 
Actually we are not able to run testcases in sync mode, we needed to add async & await to work properly.

Below is our package.json
`

>  {
>   "name": "",
>   "version": "1.0.0",
>   "private": true,
>   "description": "",
>   "devDependencies": {
>     "@rpii/wdio-report-events": "0.2.0",
>     "@wdio/cli": "7.33.0",
>     "@wdio/junit-reporter": "7.33.0",
>     "@wdio/local-runner": "7.33.0",
>     "@wdio/mocha-framework": "^7.33.0",
>     "@wdio/reporter": "7.33.0",
>     "@wdio/selenium-standalone-service": "7.33.0",
>     "@wdio/spec-reporter": "7.33.0",
>     "@wdio/sync": "^7.27.0",
>     "chai": "4.3.10",
>     "cross-env": "7.0.3",
>     "eslint": "8.10.0",
>     "eslint-plugin-wdio": "7.4.2",
>     "log4js": "6.4.1",
>     "moment": "2.29.4",
>     "prettier": "2.5.1",
>     "request": "2.88.2",
>     "request-promise": "4.2.6",
>     "source-map-support": "0.5.21",
>     "tough-cookie": "4.0.0",
>     "wdio-html-nice-reporter": "7.9.1"
>   },
>   "scripts": {
>     "lint": "eslint .",
>     "pretest-local-chrome": "npm run lint && npm run prettier:check",
>     "test-local-chrome": "cross-env SELENIUM_BROWSER=chrome wdio run wdio.conf.local.js",
>     "pretest-local-firefox": "npm run lint && npm run prettier:check",
>     "test-local-firefox": "cross-env SELENIUM_BROWSER=firefox wdio run wdio.conf.local.js",
>     "pretest-cloud": "npm run lint && npm run prettier:check",
>     "test-cloud": "wdio run wdio.conf.cloud.js",
>     "clean": "rm -rf node_modules reports",
>     "prettier:check": "prettier --check '**/*.{js,css}' --config .prettierrc",
>     "prettier:fix": "prettier --write '**/*.{js,css}' --config .prettierrc"
>   }
> }

**Other details :** 

> Node version v14.21.3, npm : 6.14.18
> chrome version: 128.0.6613.11
> Have set the sync: true in wdio.conf.js

But still my testcases are not running in sync mode. When running the below code 

> const saveButton = $('button[variant="primary"]'); 
> expect(saveButton).toBeDisplayed(); 
> Always return true if element doesn't exist too. 

So, wanted to know is there any way we can support sync mode for our testcases? 

**Note:** : Currently, we are not looking for any upgrade that why we are maintaining with the same configuration and testcases. It would be great if you can just let us know some clues on it.

**Reply by cg-tester** [reply]
Agree. @alwinjoseph02 
What is your skype?

**Comment by cg-tester** [comment]
@MahadevMG 
I am senior QA automation engineer who has long history with Webdriverio.
Let us cooperate.

**Reply by Ola-Olawoyin-sains** [reply]
Hi, I am an intermediate QA engineer using WDIO on a project for the first time... I'm having issues with mocking devtool, specifically the local storage in the application tab... any ideas please

**Reply by christian-bromann** [reply]
@Ola-Olawoyin-sains welcome 👋 join our Discord to get support on matters like this 😉

**Comment by alwinjoseph02** [comment]
Hi Team,

I was able to solve this after downgrading the "@wdio/mocha-framework" to 6.11.0 . Anyways thanks for reaching out to me.

Many thanks

From: cg-tester ***@***.***>
Date: Wednesday, 4 September 2024 at 8:22 AM
To: webdriverio/webdriverio ***@***.***>
Cc: Alwin Joseph ***@***.***>, Mention ***@***.***>
Subject: Re: [webdriverio/webdriverio] Welcome to the community, come here and say hi 👋 ! (Discussion #6217)

EXTERNAL: Use caution when clicking on links or opening attachments.



Agree. @alwinjoseph02<https://github.com/alwinjoseph02>
What is your skype?
